app/common/errors.py

```diff
@@ -0,0 +1,102 @@
+#!flask/bin/python3/
+from flask_restful import abort, request
+
+from app.models import BucketList, BucketListItem, User
+from functools import wraps
+
+"""
+Custom errors for the Restful API
+is defined here.
+"""
+
+
+def login_required(f):
+    """
+    A decorator function that checks
+    for a valid user.
+    """
+    @wraps(f)
+    def decorator(*args, **kwargs):
+        try:
+            token = request.headers.get('Token')
+            user = User.verify_auth_token(token)
+        except:
+            abort(401, message='Login is required')
+        return f(*args, **kwargs)
+    return decorator
+
+
+def invalid_id(f):
+    """A decorator function that checks
+       for a valid bucketlist id.
+    """
+    @wraps(f)
+    def decorator(*args, **kwargs):
+        bucketlist_id = kwargs.get('bucketlist_id')
+        bucketlist = BucketList.query.filter_by(id_no=bucketlist_id).first()
+        if not bucketlist:
+            abort(400, message='Bucketlist does not exist')
+        return f(*args, **kwargs)
+    return decorator
+
+
+def invalid_item_id(f):
+    """A decorator function that checks
+       for a valid bucketlist item id.
+    """
+    @wraps(f)
+    def decorator(*args, **kwargs):
+        bucketlistitem_id = kwargs.get('bucketitem_id')
+        bucketlist_item = BucketListItem.query.filter_by(id_no=bucketlistitem_id).first()
+        if not bucketlist_item:
+            abort(400, message='Bucketlist Item does not exist')
+        return f(*args, **kwargs)
+    return decorator
+
+
+custom_errors = {
```

Comment:
**Crucial(LO)** This data structure is not pythonic. Why? python in general adheres to use of snake_case, the exception being class names. Another reason why although this works, is that though you are storing your status code inside the dictionaries you still manually return them from within the code see line `27` in `register_user.py`.

A good way to think about this problem is to think in groups of possible api errors. E.g 400s are bad requests, 401s are unauthorised e.t.c  You can create a function like 
```
def bad_request(message):
    response = jsonify({'status': 'bad request', 'message': message})
    response.status_code = 400
    return response
```
and in your code call `bad_request("BukectList item does not exist")`. You can still go ahead to create your a dict for your message text if you feel that you might want to change the wording later.